File: app/src/main/java/com/mobile/android/chameapps/timebaskets/room/RulesDao.java
package com.mobile.android.chameapps.timebaskets.room;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import io.reactivex.Observable;

@Dao
public interface RulesDao {

    @Query("SELECT * FROM rules ORDER BY timestamp DESC")
    Observable<List<Item>> findAll();

    @Query("SELECT * FROM rules ORDER BY timestamp DESC LIMIT 1")
    LiveData<Item> findLast();

    @Query("SELECT * FROM rules WHERE id=:id")
    Item findById(String id);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(Item word);

    @Update
    int update(Item word);

    @Delete
    void delete(Item word);
}

File: app/src/main/java/com/mobile/android/chameapps/timebaskets/room/dao/RulesDao.kt
package com.mobile.android.chameapps.timebaskets.room.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import com.mobile.android.chameapps.timebaskets.room.enitities.Item
import io.reactivex.Observable

@Dao
interface RulesDao {
    @Query("SELECT * FROM rules ORDER BY timestamp DESC")
    fun findAll(): Observable<List<Item?>?>?

    @Query("SELECT * FROM rules ORDER BY timestamp DESC LIMIT 1")
    fun findLast(): LiveData<Item?>?

    @Query("SELECT * FROM rules WHERE id=:id")
    fun findById(id: String?): Item?

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(word: Item?): Long

    @Update
    fun update(word: Item?): Int

    @Delete
    fun delete(word: Item?)
}